pkg/platform/abstract/platform.go

```diff
@@ -108,9 +114,19 @@ func (ap *Platform) HandleDeployFunction(createFunctionOptions *platform.CreateF
 			if createFunctionOptions.FunctionConfig.Spec.RunRegistry == "" {
 				createFunctionOptions.FunctionConfig.Spec.RunRegistry = createFunctionOptions.FunctionConfig.Spec.Build.Registry
 			}
+
+			// on successful build set the source hash and timestamp of build
+			// use RandomString function which is a fast way to generate long random string
+			createFunctionOptions.FunctionConfig.Spec.SourceHash = tools.RandomString("", 32)
+			createFunctionOptions.FunctionConfig.Spec.Build.Timestamp = time.Now().Unix()
 		}
 	} else {
 
+		// no function build required and no image passed, means to use latest known image
```

Comment:
till now the reason not to build  was that a user has passed an image in the spec.
Now we allow to not pass an image and still not to rebuild if the `sourceHash` in the request is same as in function crd.
So either require a user to pass an image when the user passes `sourceHash` or to infer the image to be used. 
Chose the latter since the image to be used is easily inferred